### ISSUE TITLE
As a dev, I need the ability to add an empty placeholder to ui-tree-nodes

### ISSUE THREAD

**Issue by brandonreid** [angular-ui-tree/angular-ui-tree#570]
Would be great if there was an ability to add an empty node inside of `ui-tree-nodes`, similar to the empty placeholder that shows up when the entire tree is empty. 

I've faced this with angular ui router a number of times in different projects.
Say I have something like this:
<img width="230" alt="screen shot 2015-08-24 at 2 31 15 pm" src="https://cloud.githubusercontent.com/assets/6776415/9451585/d12ac6cc-4a6c-11e5-949c-28042bf0b209.png">

Sections can not be empty. When there are no questions in a section I show a message that says 'empty', but it's tough to drag questions into an empty section without un-intuitively dragging right. If I could add `ui-tree-nodes-empty-placeholder="true"` (or something more elegantly worded) to `ui-tree-nodes`, I could make a much better experience there.

**Comment by sathoro** [comment]
One option is to manually place an "empty" node. Then you can listen to events to remove the empty node if another node gets added to the parent or add an empty node if the last child gets removed from its parent.

For example my code looks a bit like this:

    dropped: function(event) {
        var sourceScope = event.source.nodesScope,
            destScope = event.dest.nodesScope;

        if (isColumn(sourceScope.$parent) && !sourceScope.hasChild()) {
            insertEmptyBlock(sourceScope.$parent.col);
        }

        if (isColumn(destScope.$parent)) {
            _.each(destScope.$parent.col.content, function(block, idx) {
                if (block.type == 'empty') {
                    destScope.$parent.col.content.splice(idx, 1);
                }
            });
        }
    }

**Comment by brandonreid** [comment]
@sathoro dude thanks!

So when you add the empty node, does it have a `ui-tree-node` attribute? That's needed for the drag/drop to interact with it. If so, it must be adding to the json data right? Which I just realized isn't a big deal..

**Comment by sathoro** [comment]
Yeah it needs to be a `ui-tree-node` so what I did is dynamically add it into the node array if it was otherwise empty. Then you can add `data-nodrag` to it and a custom class so you can properly style it. Then using similar code to what I posted above it will automatically be added/remove from the tree on drag/drop events.

**Comment by brandonreid** [comment]
@sathoro Thanks again! I'll let a core dev close the issue if they want. Would still be nice to have an easy way to do this that doesn't effect the data structure, but this solution does the trick.

**Comment by Voles** [comment]
Great to read you've found a solution together! :+1: 

I'm not sure how we could add this feature to the component. `ui-tree-nodes-empty-placeholder="true"` is a good suggestion. We could allow this attribute i.c.w. the `ui-tree-nodes` directive.

Any thoughts on this?

**Comment by shubhhankar** [comment]
Hey can anybody help me out how to use this angular treeview. Right now i m using ng-template myself for looping through my Json Data but i am facing many issues and the code is confusing. I am also not able to display glyphicons other than plus and minus. I want to use folder and file icon.

**Comment by Jrawke** [comment]
ui-tree-nodes-empty-placeholder="true" would be useful to me as well. It took a long time to figure out how to drag right and add to an empty node

**Comment by area** [comment]
A solution that doesn't effect the data structure would be really great from my perspective; I'm updating the data structure in real-time between all users and, significantly, my database, so adding 'fake' elements that are judiciously removed isn't really a good option.

**Comment by burakkilic** [comment]
+1
